staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/subresource.go

```diff
@@ -17,50 +17,151 @@ limitations under the License.
 package fieldmanager
 
 import (
+	"fmt"
+
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/internal"
 	"sigs.k8s.io/structured-merge-diff/v4/fieldpath"
 )
 
-// FindEntries returns all the managed fields which has a specific path
-func FindEntries(entries []metav1.ManagedFieldsEntry, pathParts ...interface{}) ([]metav1.ManagedFieldsEntry, error) {
-	path, err := fieldpath.MakePath(pathParts...)
+// SubresourceHandler manages the conversion of managed fields between a main
+// resource and subresources
+type SubresourceHandler struct {
+	parentEntries           []metav1.ManagedFieldsEntry
+	parentGroupVersion      schema.GroupVersion
+	subresourceGroupVersion schema.GroupVersion
```

Comment:
You could probably drop this then.

Reply:
You forgot to remove this :-)

Reply:
Mmmh I think we still need it, when we transform the take the managed fields of the main resource and transform them into scale we need to change there `APIVersion` into `subresourceGroupVersion` we can pass it in other ways though

Reply:
Added a field called `defaultGroupVersion` to clarify this.